enclave/core/sgx/calls.c

```diff
@@ -298,6 +341,17 @@ oe_result_t oe_handle_call_enclave_function(uint64_t arg_in)
     if (buffer)
         oe_free(buffer);
 
+    if (result != OE_OK)
+    {
+        return_args_ptr->result = result;
+        return_args_ptr->deepcopy_out_buffer = NULL;
+        return_args_ptr->deepcopy_out_buffer_size = 0;
+        memcpy(
+            args.output_buffer,
```

Comment:
Do we ensure that args.output_buffer is out side the enclave?

Reply:
The check has already been done: https://github.com/openenclave/openenclave/blob/0fe5fbc80d80f94ce0899ba8edc9133badfb1c48/enclave/core/sgx/calls.c#L233

Reply:
If that check fails and we jump to `done`, then `return_args_ptr` will be NULL and the deref will crash.

Reply:
Good catch. Fixed.